pkg/scheduler/algorithm/generic.go

```diff
@@ -58,7 +57,7 @@ const (
 var ErrNoClustersAvailable = fmt.Errorf("no clusters available to schedule subscriptions")
 
 type genericScheduler struct {
-	cache                       schedulercache.Cache
+	cache                       utils.MclsCache
```

Comment:
Don't modify this.

The cache here is for `scheduler`.

Reply:
then the same file `cache` have to write twice, I moved it to util to share between hub and scheduler